\documentclass[a4paper]{article}
\usepackage{ listings }
%\VignetteIndexEntry{Using CycleMix}
\begin{document}
\SweaveOpts{concordance=TRUE}

\title{Cell-cycle assignment using Mixture Models}
\author{Tallulah Andrews}

\maketitle

\section*{Introduction}

<<>>=
set.seed(1973)
require("SingleCellExperiment")
require("scater")
require("CycleMix")
@

\section*{Genesets}

This package provides some published cell-cycle marker gene sets for human (HGeneSets) and mouse (MGeneSets).
Sources of the gene sets can be found in the manual e.g. \texttt{?HGeneSets}.

<<>>=
names(HGeneSets)
@

You can use your own geneset simply by creating a data.frame of the same format:

<<>>=
head(MGeneSets$Cyclone)
@

The "Dir" column is used to weight the gene expression we have simplified this to positive (1) and negative (-1) markers, but any numeric value can be used, for instance fold change. 

We can convert human cell-cycle genes to mouse orthologs using \texttt{biomaRt} with our provided wrappers. For instance if we want to use the quiescence markers in mouse we would convert them as so:

<<>>=
require("biomaRt")
#map <- downloadEnsemblData()
mouse_g0 <- HGeneSets$Quiesc
#mouse_g0$Gene <- mapGeneNames(map, mouse_g0$Gene, in.name="symbol", in.org="Hsap", out.name="symbol", out.org="Mmus")
mouse_g0 <- mouse_g0[mouse_g0$Gene != "",]
@

Gene sets can also be combined together easily:

<<>>=
mouse_CC <- rbind(mouse_g0, MGeneSets$Cyclone)
@

\section*{Input Data}

CycleMix requires a SingleCellExperiment object as input. For this tutorial we have provided some example data 
already formatted correctly. Crucially there must be a column of the rowData dataframe which contains gene IDs matching your marker gene table.

<<>>= 
head(rowData(Ex))
@

You should also have a log-transformed normalized expression matrix in the SingleCellExperiment object, in our case this matrix is called "logcounts":

<<>>=
names(assays(Ex))
@

See the SingleCellExperiment or scater packages for details on creating and normalizing data in SingleCellExperiment objects.

\section*{Assigning Cells}

We can now assign cell-cycle stages to our cells. For our example data we know all the cells are cycling thus won't include the quiescence markers.

<<>>=
output <- classifyCells(Ex, MGeneSets$Cyclone)
summary(factor(output$phase))
@

Our example data comes from staged cells so we can compare our assignments to the ground truth:

<<>>=
table(factor(output$phase), Ex$cell_type1)
@

We can also examine the gaussian mixture model fit to each cell-cycle stage:
<<fig=TRUE, height=4, width=8>>=
plotMixture(output$fit[["G2M"]], BIC=TRUE)
@

The plot on the left shows the 6 different models considered : mixtures of 1-3 gaussian distributions with equal (E) or different (V) variances. The BIC criterion was used to select the optimal model. The plot on the right shows the distribution of expression scores across all cells. The curves of the fitted distributions are plotted on top. The threshold for assigning cells to the stage (if applicable) is indicated with the red dotted line. 

\end{document}
